src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.Logging
+{
+    using Microsoft.Extensions.Logging;
+
+    internal class ApplicationInsightsLoggerProvider : ILoggerProvider
```

Comment:
User would need to call `AddApplicationInsights` on `WebHostBuilder/ServiceCollection` this would add logger into `ILoggerFactory` and the next time he resolves `ILogger<T>` application insights would already be connected.